Ported/combat-bees/Assets/Scripts/Systems/BeeSpawnerSystem.cs

```diff
@@ -30,16 +36,31 @@ public void OnUpdate(ref SystemState state)
         var ecbSingleton = SystemAPI.GetSingleton<BeginSimulationEntityCommandBufferSystem.Singleton>();
         var ecb = ecbSingleton.CreateCommandBuffer(state.WorldUnmanaged);
 
-        var beeSpawnJob = new SpawnJob
+        var nestEntities = NestQuery.ToEntityArray(Allocator.Temp);
+
+        foreach (var nest in nestEntities)
         {
-            // Note the function call required to get a parallel writer for an EntityCommandBuffer.
-            ECB = ecb.AsParallelWriter(),
-            Prefab = config.bee,
-            InitTM = state.EntityManager.GetComponentData<LocalToWorldTransform>(config.bee)
-        };
+            var nestFaction = state.EntityManager.GetComponentData<Faction>(nest);
+            if (nestFaction.Value == (int)Factions.None)
+            {
+                continue;
+            }
+            
+            var nestArea = state.EntityManager.GetComponentData<Area>(nest);
+            var beeSpawnJob = new SpawnJob
+            {
+                Aabb = nestArea.Value,
+            
+                // Note the function call required to get a parallel writer for an EntityCommandBuffer.
+                ECB = ecb.AsParallelWriter(),
+                Prefab = config.bee,
+                InitFaction = state.EntityManager.GetComponentData<Faction>(nest).Value
+            };
 
-        JobHandle jobHandle = beeSpawnJob.Schedule(config.beeCount, 64);
-        jobHandle.Complete();
+            //state.Dependency = beeSpawnJob.Schedule(config.beeCount, 64, state.Dependency);
+            JobHandle jobHandle = beeSpawnJob.Schedule(config.beeCount, 64);
+            jobHandle.Complete();
```

Comment:
@hmilletUnity :
This should be ``state.Dependency = beeSpawnJob.Schedule(config.beeCount, 64, state.Dependency);``, no?

I think you forgot to reactivate the async version above.
See also the FoodSpawnerSystem, which does the right thing.